live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/messaging/config/jackson/UpdateNodeMetadataEventSerializerTest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * #%L
+ * Alfresco HX Insight Connector
+ * %%
+ * Copyright (C) 2023 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+
+package org.alfresco.hxi_connector.live_ingester.messaging.config.jackson;
+
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import static org.alfresco.hxi_connector.live_ingester.domain.model.out.PredefinedNodeProperty.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+
+import lombok.SneakyThrows;
+import org.alfresco.hxi_connector.live_ingester.domain.model.out.event.UpdateNodeMetadataEvent;
+import org.apache.commons.io.IOUtils;
+import org.junit.jupiter.api.Test;
+
+class UpdateNodeMetadataEventSerializerTest
+{
+    private final UpdateNodeMetadataEventSerializer serializer = new UpdateNodeMetadataEventSerializer();
+
+    @Test
+    public void shouldSerializeEmptyEvent()
+    {
+        UpdateNodeMetadataEvent emptyEvent = UpdateNodeMetadataEvent.create();
+
+        String expectedJson = getExpectedMapping("empty.json");
+        String actualJson = serialize(emptyEvent);
+
+        assertEquals(expectedJson, actualJson);
+    }
+
+    @Test
+    public void shouldSerializePropertiesToSet()
+    {
+        UpdateNodeMetadataEvent event = UpdateNodeMetadataEvent.create()
+                                            .set(NAME.withValue("some-name"))
+                                            .set(IS_FILE.withValue(true))
+                                            .set(MODIFIED_BY_USER_WITH_ID.withValue("000-000-000"));
+
+        String expectedJson = getExpectedMapping("with-properties-to-set.json");
+        String actualJson = serialize(event);
+
+        assertEquals(expectedJson, actualJson);
```

Comment:
Isn't this non-deterministic? Do we somehow guarantee the order of the properties?

Reply:
it is probably not guaranteed but it works ;) so for now I would leave it as it is as it is the simplest way to test it and in case of any problems we can always find some better way